doc/assessment-design.md

```diff
@@ -0,0 +1,126 @@
+# Assessment design
+
```

Comment:
Consider re-emphasizing two types: `Homework` and `Exam`. Otherwise, the later quiz portion might confuse folks since `type: 'Exam'` with `set: 'Quiz'`.